middleware/middleware.go

```diff
@@ -66,12 +67,30 @@ func ExtractGRPCAuthSubject(ctx context.Context) (string, error) {
 	return extractGRPCMetadata(ctx, metadataSubjectKey)
 }
 
+// ExtractGRPCAuthEmail extracts the custom email (email) claim from the context metadata. This claim is
+// usually injected in a middleware such as BearerToken or BearerAuthFuncGRPC, if present.
+//
+// This claim is expected in those provider that inject an email address in their JWT. Not all providers
+// do such thing.
+//
+// This function only works with gRPC requests. It returns an error if the metadata couldn't be parsed or the email
+// is not present.
+func ExtractGRPCAuthEmail(ctx context.Context) (string, error) {
+	return extractGRPCMetadata(ctx, metadataEmailKey)
+}
+
```

Comment:
Can we keep Inject/Extract functions for each field together, instead of grouping by `Extract` and `Inject` methods?